app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -262,7 +262,8 @@ def normalize_vm_attrs(vm_attrs)
 
     def url_for_fog
       u = URI.parse(url)
-      "#{u.scheme}://#{u.host}:#{u.port}"
+      path = u.path.match(/(.*)\/v\d+.*/)[1]
```

Comment:
For regexes with `/` in it I commonly use `%r{}`
```suggestion
      path = u.path.match(%r{(.*)/v\d+.*})[1]
```

I'm also wondering if this is guaranteed to only accept URLs with `/vN` in them. Otherwise this will end up raising an exception.